src/components/camera/CameraStream.jsx

```diff
@@ -13,44 +13,73 @@ import "./CameraStream.css";
  * Takes:
  *    cameraTitle: the camera title,
  *    cameraName: the camera name,
- *    unloadCallback, a callback that is ran before the window is fully unloaded
- * Returns a JSON object with keys: window, canvas, context, aspectRatio
+ *    unloadCallback: a callback that is ran before the window is fully unloaded
+ *    video_width: the stream width
+ *    video_height: the stream height
+ * Returns: Promise of an object with keys: window, canvas, context, aspectRatio
  */
-function createPopOutWindow(cameraTitle, cameraName, unloadCallback) {
-  let newWindow = window.open("", "", "width=500,height=500");
-  newWindow.document.body.style.margin = "0";
-  newWindow.document.title = cameraTitle + " Stream";
-  newWindow.document.body.innerHTML = `
-    <div style="background-color:black;width:100%;height:100%;font-family:Arial,Helvetica,sans-serif;display:flex;justify-content:center">
-      <div style="z-index:100;position:absolute;width:100%;font-size:30px;font-weight:bold;text-align:center;color:white;padding-top:5px;padding-bottom:5px;background-color:#00000066;">${cameraTitle}</div>
-      <div id="ext-fps" style="z-index:101;position:absolute;top:5;left:5;font-size:15px;font-weight:bold;color:red;">FPS: N/A</div>
-      <canvas id="ext-vid" style="z-index:1;border:none;display:block;margin:auto 0;"></canvas>
-      </div>
-    </div>`;
-  let canvas = newWindow.document.querySelector('#ext-vid');
-  let context = canvas.getContext('2d');
-  let aspectRatio = document.querySelector(`#${cameraName}-player`).videoHeight / document.querySelector(`#${cameraName}-player`).videoWidth;
+async function createPopOutWindow(cameraTitle, cameraName, unloadCallback, video_width, video_height) {
+  let newWindow = window.open("/camera/cam_popout.html", "", "width=500,height=500");
```

Comment:
Have you verified that this works both in development and production mode, i.e. when being served?

Reply:
When being served, it seems like the Router wants to remove ".html" before resolving URLs, so I changed the static popout ".html" to ".htmx."